Patient Registry System/PatientRegistrySystem.Services/UserService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using PatientRegistrySystem.DB.Contexts;
+using PatientRegistrySystem.DB.Entities;
+using PatientRegistrySystem.DB.Repos;
+using PatientRegistrySystem.Domain.Dto;
+using System;
+using System.Linq;
+
+namespace PatientRegistrySystem.Services
+{
+    public class UserService : Controller, IUserService
+    {
+        public IMapper Mapper { get; }
+        public IGenericRepository<User> userRepository { get; }
+
+        public UserService(IMapper mapper, IGenericRepository<User> UserGenericRepository)
+        {
+            Mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            userRepository = UserGenericRepository ?? throw new ArgumentNullException(nameof(UserGenericRepository));
+        }
+        //CreateUser/UpdateUser/DeleteUser/GetUser/GetAllUsers
+
+        public IActionResult GetAllUsers()
+        {
+            var usersFromRepo = userRepository.GetAll();
+            var DtoUsers = Mapper.Map<UserDto[]>(usersFromRepo);
+            if (!usersFromRepo.Any()) { return NotFound(); }
+            return Ok(DtoUsers);
+        }
+
+        public IActionResult GetUser(int id)
```

Comment:
these services shouldn't return `IActionResult`. `IActionResult` is for the Web project. Make sense to have these in the controllers.

Services should return regular objects, like: `public UserDto GetUser(int id)`.
And in the controller, you would get the data returned and wrap it with `Ok()`.

Same comment for the rest of the methods you have in the service here.